consensus/src/block_storage/sync_manager.rs

```diff
@@ -260,11 +260,15 @@ impl BlockStore {
             .iter()
             .any(|block| block.id() == highest_commit_cert.certified_block().id())
         {
+            info!(
+                "Found forked QC {}, fetching it as well",
+                highest_commit_cert
+            );
             let mut additional_blocks = retriever
                 .retrieve_block_for_qc(
                     highest_commit_cert,
                     1,
-                    highest_commit_cert.commit_info().id(),
+                    highest_commit_cert.certified_block().id(),
```

Comment:
this is the actual fix. we were passing the wrong id here, making the function panic.

Reply:
Are any of the unit tests able to repro this before the change?

Reply:
yes! if I don't change this line, test panics, just like the production

Reply:
For all three newly added tests?

Reply:
nono, just forking_retrieval_test.

rest are tests for the changes in the testing framework/utilities/testing things we were not testing before - like block retrieval.